meilisearch-release-changelog/v1.0.0.md

```diff
@@ -0,0 +1,205 @@
+* When calling `POST /indexes` and `PUT /indexes/:uid`
+  - When `uid` is missing in the payload, Meilisearch returns `missing_index_uid` instead of `bad_request`
+  - Sending a value with a different type than `string` or `null` for `primaryKey` now returns `invalid_index_primary_key` instead of `bad_request`
+* When calling `GET /index/:uid/documents` and `GET /indexes/:uid/documents/:uid`
+  - Using the `fields` parameter wrongly, Meilisearch now returns `invalid_document_fields` instead of `bad_request`
+  - Using the `limit` parameter wrongly, Meilisearch now returns `invalid_document_limit` instead of `bad_request`
+  - Using the `offset` parameter wrongly, Meilisearch now returns `invalid_document_offset` instead of `bad_request`
+* When calling `POST /index/:uid/documents` and `PUT /indexes/:uid/documents`
+  - Using the `primaryKey` parameter wrongly, Meilisearch now returns ` invalid_index_primary_key` instead of `bad_request`
+  - When adding a document with a wrongly formatted `_geo` field, Meilisearch now returns `invalid_document_geo_field` instead of `invalid_geo_field`[RENAME]
+* When calling `GET /index/:uid/search` and `POST /indexes/:uid/search`
+  - Using the `q` parameter wrongly, Meilisearch now returns `invalid_search_q` instead of `bad_request`
+  - Using the `offset` parameter wrongly, Meilisearch now returns `invalid_search_offset` instead of `bad_request`
+  - Using the `limit` parameter wrongly, Meilisearch now returns `invalid_search_limit` instead of `bad_request`
+  - Using the `page` parameter wrongly, Meilisearch now returns `invalid_search_page` instead of `bad_request`
+  - Using the `hits_per_page` parameter wrongly, Meilisearch now returns `invalid_search_hits_per_page` instead of `bad_request`
+  - Using the `attributesToRetrieve` parameter wrongly, Meilisearch now returns `invalid_search_attributes_to_retrieve` instead of `bad_request`
+  - Using the `attributesToCrop` parameter wrongly, Meilisearch now returns `invalid_search_attributes_to_crop` instead of `bad_request`
+  - Using the `showMatchesPosition` parameter wrongly, Meilisearch now returns `invalid_search_show_matches_position` instead of `bad_request`
+  - Using the `filter` parameter wrongly, Meilisearch now returns `invalid_search_filter` instead of `bad_request`
+  - Using the `sort` parameter wrongly, Meilisearch now returns `invalid_search_sort` instead of `bad_request`
+  - Using the `facets` parameter wrongly, Meilisearch now returns `invalid_search_facets` instead of `bad_request`
+  - Using the `highlightPreTag` parameter wrongly, Meilisearch now returns `invalid_search_highlight_pre_tag` instead of `bad_request`
+  - Using the `highlightPostTag` parameter wrongly, Meilisearch now returns `invalid_search_highlight_post_tag` instead of `bad_request`
+  - Using the `matchingStrategy` parameter wrongly, Meilisearch now returns `invalid_search_matching_strategy` instead of `bad_request`
```

Comment:
```suggestion
  - Incorrect use of the `q` parameter now returns `invalid_search_q` instead of `bad_request`
  - Incorrect use of the `offset` parameter now returns `invalid_search_offset` instead of `bad_request`
  - Incorrect use of the `limit` parameter now returns `invalid_search_limit` instead of `bad_request`
  - Incorrect use of the `page` parameter now returns `invalid_search_page` instead of `bad_request`
  - Incorrect use of the `hits_per_page` parameter now returns `invalid_search_hits_per_page` instead of `bad_request`
  - Incorrect use of the `attributesToRetrieve` parameter now returns `invalid_search_attributes_to_retrieve` instead of `bad_request`
  - Incorrect use of the `attributesToCrop` parameter now returns `invalid_search_attributes_to_crop` instead of `bad_request`
  - Incorrect use of the `showMatchesPosition` parameter now returns `invalid_search_show_matches_position` instead of `bad_request`
  - Incorrect use of the `filter` parameter now returns `invalid_search_filter` instead of `bad_request`
  - Incorrect use of the `sort` parameter now returns `invalid_search_sort` instead of `bad_request`
  - Incorrect use of the `facets` parameter now returns `invalid_search_facets` instead of `bad_request`
  - Incorrect use of the `highlightPreTag` parameter now returns `invalid_search_highlight_pre_tag` instead of `bad_request`
  - Incorrect use of the `highlightPostTag` parameter now returns `invalid_search_highlight_post_tag` instead of `bad_request`
  - Incorrect use of the `matchingStrategy` parameter now returns `invalid_search_matching_strategy` instead of `bad_request`
```